src/tests/test_raft_repl_dev.cpp

```diff
@@ -269,7 +269,12 @@ class TestReplicatedDB : public homestore::ReplDevListener {
                    "[Replica={}] Save logical snapshot callback obj_id={} term={} idx={} is_last={} num_items={}",
                    g_helper->replica_num(), snp_data->offset, s->get_last_log_term(), s->get_last_log_idx(),
                    snp_data->is_last_obj, num_items);
-        snp_data->offset = last_committed_idx + 1;
+
+        // before we finish install snapshot, raft_server()->get_committed_log_idx() will always be the same. so we need
+        // last_data_committed_lsn to notify leader to transfer new data to follower.
+        // actually, leader will transfer all the data in one shot, so this is will not trigger any error, but we should
+        // make the logic correct.
+        snp_data->offset = last_data_committed_lsn + 1;
```

Comment:
suggesting `snapshot_write_to_lsn` instead of `last_data_committed_lsn`

Reply:
this is will be updated in on_commit